### PR TITLE
[Feat] Able to overwrite custom theme

### PR DESCRIPTION
**Pull request by macrigiuseppe**
This pull request will make changes to kepler-gl.js component to allow develoeprs to overwrite Kepler theme through styled-components.

I also created an example and this is a simple result:
screen shot 2019-03-03 at 8 21 54 pm

### CODE REVIEW

**Comment by macrigiuseppe** [comment]
Update code:
- Now we use selector to generate the custom theme in kepler-gl.js component.

New result:
Screen Shot 2019-03-25 at 11 44 43 AM

Interaction
custom-theme

File: examples/custom-theme/src/app.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  render() {
+    const {width, height, customTheme} = this.state;
+    return (
+      <div>
+        <StyleSwitch>
+          <label htmlFor="custom-theme">Custom theme</label>
+          <input type="checkbox" checked={customTheme} id="custom-theme" onChange={this._onToggleTheme}/>
+        </StyleSwitch>
+        <ThemeProvider theme={customTheme ? theme : emptyTheme}>
```
Comment (heshan0131) [inline examples/custom-theme/src/app.js]: why do user has to use `ThemeProvider` ? Can't they just pass `theme ` as prop to `KeplerGl`?

Comment (macrigiuseppe) [reply examples/custom-theme/src/app.js]: yes of course. Passing the theme directly to kepler.

It works both using props and theme provider

File: yarn.lock [hunk 2]
```diff
@@ -7012,6 +7024,17 @@ react-virtualized-auto-sizer@^1.0.2:
   version "1.0.2"
   resolved "https://registry.yarnpkg.com/react-virtualized-auto-sizer/-/react-virtualized-auto-sizer-1.0.2.tgz#a61dd4f756458bbf63bd895a92379f9b70f803bd"
 
+react-virtualized@^9.21.0:
```
Comment (heshan0131) [inline yarn.lock]: should this be reverted if we are not adding react-virtualized?

File: package.json [hunk 3]
```diff
@@ -108,6 +109,7 @@
     "react-palm": "^1.1.2",
     "react-redux": "^4.0.0",
     "react-tooltip": "^3.10.0",
+    "react-virtualized": "^9.21.0",
```
Comment (heshan0131) [inline package.json]: where are we using react-virtualized in the source code?

File: README.md [hunk 4]
```diff
@@ -320,8 +320,10 @@ const MapContainer = ({dispatch}) => (
 ```
 Read more about [forward dispatching actions][forward-actions]
 
+### 4. Customize style.
+TBD
```
Comment (heshan0131) [inline README.md]: Add here?

Comment (heshan0131) [reply README.md]: Also add `theme` prop to 2. Mount Component section

File: examples/custom-theme/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Demo App
```
Comment (heshan0131) [inline examples/custom-theme/README.md]: You might want to update this doc
Also add an entry in `examples/README.md`

File: package.json (hunk 3 above)
Comment (macrigiuseppe) [reply package.json]: i think i accidentaly ran yarn add react-virtualized in the root folder

File: yarn.lock (hunk 2 above)
Comment (macrigiuseppe) [reply yarn.lock]: yep

File: README.md (hunk 4 above)
Comment (macrigiuseppe) [reply README.md]: done

File: examples/custom-theme/README.md (hunk 5 above)
Comment (macrigiuseppe) [reply examples/custom-theme/README.md]: done

File: README.md [hunk 6]
```diff
@@ -320,14 +327,102 @@ const MapContainer = ({dispatch}) => (
...
+
+
+The available properties to customize are listed here [theme](https://github.com/uber/kepler.gl/blob/master/src/styles/base.js).
+
+[Custom theme example](https://github.com/uber/kepler.gl/tree/master/examples/custom-theme).
+
+#### Passing a Theme prop.
+You can customize Kepler.gl theme by passing a __theme__ props to Kepler.gl react component as it follows:
+```javascript
+// Properties are defined in https://github.com/uber/kepler.gl/blob/master/src/styles/base.js
```
Comment (heshan0131) [inline README.md]: this is important information, should not be listed in a comment

File: README.md [hunk 7]
```diff
@@ -320,14 +327,102 @@ const MapContainer = ({dispatch}) => (
...
+const white = '#ffffff';
+const customTheme = {
```
Comment (heshan0131) [inline README.md]: nitpick, as this is a code example, we don't have to list 20+ custom themes here, 2-3 would be enough, for easy understanding

File: README.md (hunk 6 above)
Comment (macrigiuseppe) [reply README.md]: I will remove it.

The same information is listed on line 336 of this file

File: README.md (hunk 7 above)
Comment (macrigiuseppe) [reply README.md]: sounds good

**Comment by macrigiuseppe** [comment]
@heshan0131 updated with your comments

**Comment by briantcl** [comment]
finished at Sprint Mar 22, 2019

**Pull request merged**